### PR TITLE
BookDetail test

### PR DESCRIPTION
**Pull request by VoytovychM**

### CODE REVIEW

File: src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class BookDetailServiceTest {
+    private static BookDetailService bookDetailService;
+    private static BookDetailRepository bookDetailRepository;
+    private static BookRepository bookRepository;
+    private static BookService bookService;
```
Comment (IliaIliukovich) [inline src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java]: Why do you need this field?

File: src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+        oldBookDetail.setId(44L);
+
+        Mockito.when(bookDetailRepository.findById(oldBookDetail.getId())).thenReturn(Optional.of(oldBookDetail));
+        Mockito.when(bookDetailRepository.save(newBookDetail)).thenReturn(newBookDetail);
+
+        Boolean result = bookDetailService.updateBookDetail(newBookDetail);
+
+        Mockito.verify(bookDetailRepository).findById(newBookDetail.getId());
+        Mockito.verify(bookDetailRepository).save(newBookDetail);
+        assertEquals(true, result);
```
Comment (IliaIliukovich) [inline src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java]: Better to use assertTrue()

File: src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+        });
+        assertEquals("Book with id 88 not found", exception.getMessage());
+        Mockito.verify(bookRepository).findById(bookId);
+        Mockito.verify(bookDetailRepository, never()).save(any(BookDetail.class));
+
+
+    }
+
+    @Test
+    public void updateBookDetail() {
```
Comment (IliaIliukovich) [inline src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java:92]: return false case should also be checked

File: src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java (hunk 1 above)
Comment (VoytovychM) [reply src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java]: I have deleted bookService field

File: src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java (hunk 2 above)
Comment (VoytovychM) [reply src/test/java/com/telran/libraryapp/service/BookDetailServiceTest.java]: Done

**Review by IliaIliukovich — APPROVED** [review]

**Pull request merged**